### PR TITLE
Refactor2 hl

### PR DESCRIPTION
**Pull request by hale4029**
computer board refactored to be dynamic.
refactored board to be dynamic

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -2,23 +2,32 @@
...
   end
 
   def valid_placement?(ship_name, coordinates_array)
     horizontal = coordinates_array.map { |x| x.delete("ABCDEFGHIJKLMNOPQRSTUVWXYZ").to_i }
     vertical = coordinates_array.map { |coordinate| coordinate.delete("1234567890") }
     vertical = vertical.map! { |letter| letter.ord }
-    ((1..4).each_cons(ship_name.length).to_a.include?(horizontal) && vertical.uniq.length == 1) ||
-    ((65..68).each_cons(ship_name.length).to_a.include?(vertical) && horizontal.uniq.length == 1)
+    ((1..(1 + (@dimensions - 1))).each_cons(ship_name.length).to_a.include?(horizontal) && vertical.uniq.length == 1) ||
+    ((65..(65 + (@dimensions - 1))).each_cons(ship_name.length).to_a.include?(vertical) && horizontal.uniq.length == 1)
```
Comment (dylanconnolly) [inline lib/board.rb:30]: starting the each_cons array at 65 may cause errors for ship placement validation if the user inputs coordinates that don't follow [letter][number][letter][number] (A7A9A0). For example, if they input a string of just numbers (12301923) will the ordinal exist in that range of (65..(dimensions - 1))?

File: test/computer_board_test.rb [hunk 2]
```diff
@@ -3,28 +3,23 @@
...
+    @computer.computer_board.create_cells
   end
 
   def test_it_exists
     assert_instance_of ComputerBoard, @computer
   end
 
   def test_placement_of_ships
-    @computer.place_cruiser
+    @computer.place_ships
```
Comment (dylanconnolly) [inline test/computer_board_test.rb:20]: do we create the ships here or in the game file based off the amount of ships the user chooses to create?

File: lib/board.rb (hunk 1 above)
Comment (hale4029) [reply lib/board.rb:30]: We'll see..

File: test/computer_board_test.rb (hunk 2 above)
Comment (hale4029) [reply test/computer_board_test.rb:20]: here.

**Review by dylanconnolly — APPROVED** [review]

**Pull request merged**